src/inmanta/deploy/scheduler.py

```diff
@@ -1125,9 +1131,16 @@ async def _update_scheduler_state_for_finished_deploy(
                 # except for the event to notify dependents of failure recovery (to unblock skipped for dependencies)
                 # because we might otherwise miss the recovery (in the sense that the next deploy wouldn't be a transition
                 # from a bad to a good state, since we're transitioning to that good state now).
+
+                # The second part of the or would not be required because is implied by the first,
+                # except that we don't enforce the hash diff.
+                # We emit a warning if we observe this, but that still doesn't prevent it.
+                # While it should not happen it can
                 state.deployment_result = deployment_result
+                state.last_deployed = finished
                 if recovered_from_failure:
                     self._send_events(details, stale_deploy=True, recovered_from_failure=True)
+                self._timer_manager.update_timer(resource, state=state)
```

Comment:
I wonder if we might want to move those to `deploy_done()` under the `finally` for robustness. Wdyt?

Speaking of, you raised the issue of `work.finished_deploy()` needing to be in a `finally` block. I assumed that meant you did it in this PR but I don't see it. Do you plan to add it or do we create a separate PR? Just so we don't lose track of it.

Reply:
done both